phygnn/model_interfaces/base_model.py

```diff
@@ -802,30 +889,29 @@ def _parse_features(self, features, names=None, process_one_hot=True,
             logger.error(msg)
             raise RuntimeError(msg)
 
-        if self.feature_names is not None:
-            if features.shape[1] != len(self.feature_names):
-                msg = ('data has {} features but expected {}'
-                       .format(features.shape[1], self.feature_dims))
-                logger.error(msg)
-                raise RuntimeError(msg)
-
-        if self._feature_names is None:
+        if self.feature_names is None:
             self._feature_names = feature_names
+
+        if self.one_hot_categories is not None:
```

Comment:
So OHE only gets run if OHE_categories is input right? Should clarify in doc strings.

Reply:
Yes, will do